pkg/virt-controller/services/template.go

```diff
@@ -1474,10 +1474,16 @@ func (t *templateService) VMIResourcePredicates(vmi *v1.VirtualMachineInstance,
 
 	metrics.SetVmiLaucherMemoryOverhead(vmi, memoryOverhead)
 	withCPULimits := t.doesVMIRequireAutoCPULimits(vmi)
+	additionalCPUs := uint32(0)
+	if vmi.Spec.Domain.IOThreads != nil &&
+		vmi.Spec.Domain.IOThreads.Policy == v1.IOThreadsPolicyDedicated &&
+		vmi.Spec.Domain.IOThreads.Count != nil {
+		additionalCPUs = *vmi.Spec.Domain.IOThreads.Count
+	}
```

Comment:
Some technical questions:
1. Is it a requirement that each iothread would need a whole dedicated CPU to run on? In the worst case where all of the iothreads would be busy they would have to compete for CPU time, but is this case common? Should we  for example consider to allocate only 1 additional CPU for all of the iothreads?
2. Is it reasonable to pin the iothreads to the "housekeeping" CPU? Generally this CPU is assumed to not be very busy, and its main job is to isolate "housekeeping" tasks (which aren't CPU bound) from the pCPUs that are used exclusively by vCPUs.

Reply:
> Some technical questions:
> 
> 1. Is it a requirement that each iothread would need a whole dedicated CPU to run on? In the worst case where all of the iothreads would be busy they would have to compete for CPU time, but is this case common? Should we  for example consider to allocate only 1 additional CPU for all of the iothreads?

I've been talking with @stefanha about this, and yes each iothread should be pin to a separate pCPU. Otherwise, just use a single iothread. @stefanha please correct me here if I'm wrong.

> 2. Is it reasonable to pin the iothreads to the "housekeeping" CPU? Generally this CPU is assumed to not be very busy, and its main job is to isolate "housekeeping" tasks (which aren't CPU bound) from the pCPUs that are used exclusively by vCPUs.

Well, this is how it is done today. If somebody doesn't want to allocate those extra cpus then they can go with the options we already have.

Reply:
> I've been talking with @stefanha about this, and yes each iothread should be pin to a separate pCPU. Otherwise, just use a single iothread. @stefanha please correct me here if I'm wrong.

Yes.